appserver/data/data-core/src/main/java/fish/payara/data/core/util/QueryByNameOperationUtility.java

```diff
@@ -228,59 +245,77 @@ private static jakarta.persistence.Query buildQueryFromParser(QueryMethodParser
         Map<String, String> joinAliases = new HashMap<>();
 
         buildQueryClause(jpql, executionAction, dataForQuery, rootAlias);
+
+        List<Sort<?>> dynamicSorts = collectSortsFromArgs(dataForQuery.getMethod(), args);
```

Comment:
if you define the reference of the DataParameter, you also will have the sortlist available from the args, you don't need to call the collectSortsFromArgs, we need to see if this is needed because we have duplicated functionality
```suggestion
        List<Sort<?>> dynamicSorts = parameter.sortList();
```